index.js

```diff
@@ -18,20 +18,41 @@ module.exports = function (cmd, args, opts) {
 				val = stripEof(val);
 			}
 
+			if (parsed.options.encoding !== undefined) {
+				val = new Buffer(val, parsed.options.encoding);
+			}
+
 			return val;
 		};
 
-		var spawned = childProcess.execFile(parsed.command, parsed.args, parsed.options, function (err, stdout, stderr) {
```

Comment:
My last sentence makes no sense. We of course don't want to execute it twice for a command. I think I meant to say that `execFile` returns the child process. So could maybe listen to .stdout/.stderr and gather it and then include it in the `execFile` callback.

Maybe @jamestalmage has a better suggestion.